tests/test_core/test_storage.py

```diff
@@ -68,6 +68,36 @@ def test_get_custom_folder(local_storage_manager: StorageManager, project_folder
     assert local_storage_manager.get_folder("eopatches", full_path=True) == abs_path
 
 
+@pytest.mark.parametrize("config_profile", [None, "", "nonexistent-config-profile"])
+@pytest.mark.parametrize("env_profile", [None, "", "nonexistent-env-profile"])
+def test_aws_profile(aws_storage_config: RawConfig, config_profile: Optional[str], env_profile: Optional[str]):
+    """Checks different combinations of profile being set with a config parameter and environmental variable. Checks
+    also that config parameter takes priority over environmental variable."""
+
+    for parameter_key, parameter_value, config_dict in [
+        ("aws_profile", config_profile, aws_storage_config),
+        ("AWS_PROFILE", env_profile, os.environ),
+    ]:
+        if parameter_value is not None:
+            config_dict[parameter_key] = parameter_value
+        elif parameter_key in config_dict:
+            del config_dict[parameter_key]
+
+    try:
+        expected_profile = config_profile if config_profile is not None else env_profile
+        if expected_profile:
+            with pytest.raises(ProfileNotFound) as exception_info:
+                StorageManager.from_raw_config(aws_storage_config)
+
+            assert str(exception_info.value) == f"The config profile ({expected_profile}) could not be found"
```

Comment:
isn't it odd that we have an `expected_profile` despite expecting the profile not to exist?

Reply:
Not really. The way this works is:

1. some profile name is passed from config or env variable to `StorageManager`,
2. `StorageManager` uses an utility that uses `botocore` to try load credentials from `~/.aws/credentials` that are saved under given profile name.
3. Because the profile name doesn't exist we get `ProfileNotFound` exception.
4. The test checks the exception's error message to see if the name of the profile it had to load is the one that we expected it to try to load.